### PR TITLE
Notification View for WorksForYouVC

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] Created a class that allows the WorksForYouVC to present artist and artwork VCs when tapped
- [x] Mark notifications as read in viewWillAppear
- [x] Tests

I can add all the UI tinkering to this, but I'm thinking of putting that in its own PR and leaving this as a more functionality-focused one.

### CODE REVIEW

**Comment by sarahscott** [comment]
notificationview

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">1 Error</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:no_entry_sign:</td>
      <td><p>No CHANGELOG changes made</p>
</td>
    </tr>
    
  </tbody>
</table>
  

  
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%">3 Warnings</th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td><p>PR is classed as Work in Progress</p>
</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td><p>Needs testing on a Phone if change is non-trivial</p>
</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td><h4>Found unstubbbed networking requests</h4>

<ul>
<li><a href="https://github.com/search?q=ARArtworkFavoritesNetworkModel+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARArtworkFavoritesNetworkModelSpec</a> in <code>test_getFavorites__makes_request_if_no_request_is_in_progress</code></li>
<li><a href="https://github.com/search?q=ARArtworkViewControllerBuyButton+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARArtworkViewControllerBuyButtonSpec</a> in <code>test_buy_button__posts_order_if_artwork_has_no_edition_sets</code></li>
<li><a href="https://github.com/search?q=ARCreateAccountViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARCreateAccountViewControllerSpec</a> in <code>test_warning_view__displays_a_message</code></li>
<li><a href="https://github.com/search?q=ARCreateAccountViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARCreateAccountViewControllerSpec</a> in <code>test_warning_view__displays_a_message</code></li>
<li><a href="https://github.com/search?q=AREmbeddedModelPreviewViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">AREmbeddedModelPreviewViewControllerSpec</a> in <code>test_iPhone_peek__shows_favorite_and_follow_buttons_for_logged_in_user</code></li>
<li><a href="https://github.com/search?q=AREmbeddedModelPreviewViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">AREmbeddedModelPreviewViewControllerSpec</a> in <code>test_iPhone_peek__shows_favorite_and_follow_buttons_for_logged_in_user</code></li>
<li><a href="https://github.com/search?q=AREmbeddedModelPreviewViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">AREmbeddedModelPreviewViewControllerSpec</a> in <code>test_iPhone_peek__passes_artwork_to_share_controller</code></li>
<li><a href="https://github.com/search?q=ARFairViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARFairViewControllerSpec</a> in <code>test_creates_an_NSUserActivity</code></li>
<li><a href="https://github.com/search?q=ARGeneViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARGeneViewControllerSpec</a> in <code>test_creates_an_NSUserActivity</code></li>
<li><a href="https://github.com/search?q=ARGeneViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARGeneViewControllerSpec</a> in <code>test_creates_an_NSUserActivity</code></li>
<li><a href="https://github.com/search?q=ARTopMenuViewController+repo%3Aartsy%2Feigen&amp;ref=searchresults&amp;type=Code&amp;utf8=%E2%9C%93">ARTopMenuViewControllerSpec</a> in <code>test_navigation__favorites__logged_in__tab_behavior__already_selected__pops_to_root</code></li>
</ul>
</td>
    </tr>
    
  </tbody>
</table>
  

  


<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/KrauseFx/danger/">danger</a>
</p>

File: Artsy/View_Controllers/WorksForYou/ARWorksForYouViewController.m [hunk 1]
```diff
@@ -47,15 +53,7 @@ - (void)viewDidAppear:(BOOL)animated
...
-
-    self.view.backgroundColor = [UIColor whiteColor];
-    self.view.directionalLockEnabled = YES;
-    self.view.delegate = self;
-
-    _worksForYouNetworkModel = [[ARWorksForYouNetworkModel alloc] init];
-
-    [self.view.stackView alignLeading:@"20" trailing:@"20" toView:self.view];
-    [self.view.stackView alignTopEdgeWithView:self.view predicate:@"20"];
+    _worksForYouNetworkModel = _worksForYouNetworkModel ?: [[ARWorksForYouNetworkModel alloc] init];
```
Comment (alloy) [inline Artsy/View_Controllers/WorksForYou/ARWorksForYouViewController.m]: Use the property setter/getter method outside of `init` / `dealloc`.

File: Artsy/View_Controllers/WorksForYou/ARWorksForYouViewController.m [hunk 2]
```diff
@@ -172,6 +129,11 @@ - (void)markNotificationsAsRead
     // will go in network model
 }
 
+- (void)artistTapped:(UIGestureRecognizer *)recognizer
+{
+    //    recognizer
+}
+
```
Comment (alloy) [inline Artsy/View_Controllers/WorksForYou/ARWorksForYouViewController.m]: What is this for?

File: Artsy/View_Controllers/WorksForYou/ARWorksForYouViewController.m [hunk 3]
```diff
@@ -138,8 +85,18 @@ - (void)addSeparatorLine
 - (void)addNotificationItems:(NSArray *)items
 {
     [items each:^(ARWorksForYouNotificationItem *item) {
-        UIView *viewItem = [self viewBasedOnNotificationItem:item];
-        [self.view.stackView addSubview:viewItem withTopMargin:@"0" sideMargin:@"20"];
+        
+        // since this view controller will be handling all navigation, it should be the embedded VC delegate
+        AREmbeddedModelsViewController *worksVC = [[AREmbeddedModelsViewController alloc] init];
+        worksVC.delegate = self;
```
Comment (alloy) [inline Artsy/View_Controllers/WorksForYou/ARWorksForYouViewController.m]: The embedded models view controller is never added as a child VC to any parent VC, which means it won’t receive rotation events (on iPad) and possibly other issues.

But adding the embedded models VC to the VC of this view and then using the embedded models VC’s view in the item view feels a bit weird. I think the ‘item’ should be responsible for all.

So maybe you need a `ARWorksForYouNotificationItemViewController` which as a child has the embedded models VC and then add the item VC as a child VC of this view’s VC.

The delegate stuff I think you could consolidate to the new VC by making your delegate protocol also include the embedded VC’s one. E.g.

```objc
@protocol ARWorksForYouNotificationItemViewControllerDelegate <AREmbeddedModelsViewControllerDelegate>
- (void)didSelectArtist:(Artist *)artist;
@end
```

And then assign your delegate to the embedded models VC when you receive it, e.g.:

```objc
@implementation ARWorksForYouNotificationItemViewController

- (void)setDelegate:(id< ARWorksForYouNotificationItemViewControllerDelegate>)delegate
{
    _delegate = delegate;
    self.artworksVC.delegate = (id<AREmbeddedModelsViewControllerDelegate>)delegate;
}

@end
```

Comment (orta) [inline Artsy/View_Controllers/WorksForYou/ARWorksForYouViewController.m]: The VC is being added: `+        [self addViewController:self.artworksVC toParent:(id)self.artworksVC.delegate withTopMargin:@"10" sideMargin:@"0"];`

File: Artsy/View_Controllers/WorksForYou/ARWorksForYouViewController.m (hunk 2 above)
Comment (sarahscott) [inline Artsy/View_Controllers/WorksForYou/ARWorksForYouViewController.m]: lol this should have been deleted

**Comment by orta** [comment]
Might be worth learning how to squash these commits to re-do the wording: 

screen shot 2016-03-02 at 4 22 30 pm

They provide very little value when debugging in the future

**Comment by orta** [comment]
:+1:

**Comment by sarahscott** [comment]
i can do that

**Comment by orta** [comment]
Me + Alloy are good for this one!

**Comment by orta** [comment]
Well done

**Pull request merged**